File: backend-spring/application/src/main/java/apptemplate/application/mappers/NotificationMapper.java
package apptemplate.application.mappers;

import apptemplate.application.dto.notification.NotificationDto;
import apptemplate.domain.entities.Notification;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

@Mapper(componentModel = "spring")
public interface NotificationMapper {

    NotificationMapper INSTANCE = Mappers.getMapper(NotificationMapper.class);

    @Mapping(target = "type", expression = "java(notification.getType().name())")
    @Mapping(target = "isRead", source = "read")
    NotificationDto toDto(Notification notification);
}

File: backend/nestjs/clean-architecture/full/src/modules/notification/application/mappers/notification.mapper.ts
import { Notification } from '../../domain/entities/notification.entity';
import { NotificationDto } from '../dto/notification.dto';

export class NotificationMapper {
  static toDto(notification: Notification): NotificationDto {
    return new NotificationDto({
      id: notification.id,
      title: notification.title,
      message: notification.message,
      type: notification.type,
      referenceId: notification.referenceId,
      referenceType: notification.referenceType,
      isRead: notification.isRead,
      createdAt: notification.createdAt,
    });
  }

  static toDtoList(notifications: Notification[]): NotificationDto[] {
    return notifications.map((notification) => NotificationMapper.toDto(notification));
  }
}
